### PR TITLE
Add tests for MAPI organization listing and creating

### PR DESCRIPTION
**Pull request by kuosandys**
This PR adds some test cases for listing organizations and creating new organizations in the MAPI UI.

### CODE REVIEW

File: src/components/MAPI/organizations/__tests__/OrganizationList.tsx [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+    fireEvent.change(nameInput, {
+      target: { value: 'not-created-org' },
+    });
+
+    const cancelButton = screen.getByRole('button', {
+      name: 'Cancel',
+    });
+    fireEvent.click(cancelButton);
+
+    expect(screen.queryByText('not-created-org')).not.toBeInTheDocument();
```
Comment (axbarsan) [inline src/components/MAPI/organizations/__tests__/OrganizationList.tsx:275]: Is this to check if the success message is not present?

File: src/components/MAPI/organizations/__tests__/OrganizationList.tsx [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+  });
+
+  it('displays an error if an organization cannot be created', async () => {
+    nock(window.config.mapiEndpoint)
+      .post('/apis/security.giantswarm.io/v1alpha1/organizations/')
+      .reply(StatusCodes.InternalServerError, {
+        apiVersion: 'security.giantswarm.io/v1alpha1',
+        kind: 'Status',
+        message: 'Some error occurred',
+      });
```
Comment (axbarsan) [inline src/components/MAPI/organizations/__tests__/OrganizationList.tsx:297]: Not exactly the right response payload, but it works. This is how a `Status` response looks like

```typescript
{
  apiVersion: 'v1',
  kind: 'Status',
  message: 'Some error occured',
  status: metav1.K8sStatuses.Failure,
  reason: metav1.K8sStatusErrorReasons.NotFound,
  code: StatusCodes.NotFound,
}
```

**Review by axbarsan — APPROVED** [review]
Sweeet 👍

File: src/components/MAPI/organizations/__tests__/OrganizationList.tsx (hunk 1 above)
Comment (kuosandys) [reply src/components/MAPI/organizations/__tests__/OrganizationList.tsx:275]: I was thinking to check that the user input organization name is not present anywhere (both the input field and the success message). Should this be two separate checks actually?

Comment (axbarsan) [reply src/components/MAPI/organizations/__tests__/OrganizationList.tsx:275]: For the input field value there should be an extra one (i.e. `screen.queryByDisplayValue()`)

File: src/components/MAPI/organizations/__tests__/OrganizationList.tsx [hunk 3]
```diff
@@ -272,14 +273,19 @@ describe('OrganizationIndex', () => {
...
         name: 'Create organization',
       })
     ).not.toBeInTheDocument();
 
+    expect(screen.queryByText('not-created-org')).not.toBeInTheDocument();
+    await waitFor(() =>
+      expect(
+        screen.queryByDisplayValue('not-created-org')
+      ).not.toBeInTheDocument()
+    );
```
Comment (axbarsan) [inline src/components/MAPI/organizations/__tests__/OrganizationList.tsx:287]: Just for the future: there's `waitForElementToBeRemoved()`, which is made specifically for this use case

**Pull request merged**